include/fiction/layouts/coordinates.hpp

```diff
@@ -145,6 +147,39 @@ struct ucoord_t
     {
         return ucoord_t{static_cast<uint64_t>(*this) | static_cast<uint64_t>(ucoord_t{})};
     }
+    /**
+     * Clamps the coordinate to the given dimension. Lower bounds are neglected and should be handled separately if need
+     * be. When the coordinate is outside the given dimension, it is transformed to the next coordinate that is in it.
+     * If no such coordinate exists, the coordinate becomes a dead copy of the dimension.
+     *
+     * @param dimension Dimension or aspect ratio to clamp the coordinate to.
+     */
```

Comment:
I don't think I understand the docstring text yet. Could you maybe rephrase it and also add a quick note on the use case of this function?